docs/ref/databases.txt

```diff
@@ -244,6 +244,34 @@ database configuration in :setting:`DATABASES`::
         },
     }
 
+.. _postgresql-pool:
+
+Connection Pool
+---------------
+
+To use a connection pool with `psycopg`_ 3.1.8+, you can either set ``"pool"``
+in the :setting:`OPTIONS` part of your database configuration in
+:setting:`DATABASES` to be a dict to be passed to
+:class:`~psycopg:psycopg_pool.ConnectionPool` or to ``True`` to use the
+``ConnectionPool`` defaults::
+
+    DATABASES = {
+        "default": {
+            "ENGINE": "django.db.backends.postgresql",
+            # ...
+            "OPTIONS": {
+                "pool": True,
+            },
+        },
+    }
+
+This option requires ``psycopg[pool]`` or ``psycopg_pool`` to be installed and
+is ignored with ``psycopg2``.
+
+.. warning::
+
+    Connection pools do not handle dynamically updated database settings.
```

Comment:
As far as I'm aware, this is generally not supported (and never has been). I don't think we need this warning.